### PR TITLE
kube3d: 3.0.2 -> 3.1.0 -> 3.1.1 -> 3.1.2 -> 3.1.3 -> 3.1.4 -> 3.1.5

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Motivation for this change

Bump kube3d/k3d to ~`3.1.0`~ ~`3.1.1`~ ~`3.1.2`~ ~`3.1.3`~ ~`3.1.4`~ `3.1.5`

###### Things done

Updated version, k3s version, and src sha256
I matched the k3s version to the one output by the GH release binary, hopefully that's correct.

@kuznero @jlesquembre @ngerstle 

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS (x86_64)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
Friendly bump @kuznero @jlesquembre @ngerstle

**Comment by 06kellyjac** [comment]
I've rebased on master & corrected my mistake with the k3sVersion having the `v` attached, also version bumped again

**Comment by jlesquembre** [comment]
@06kellyjac Thanks for your PR, I didn't have time to test it, but since it's a simple version dump , it should be ok

**Comment by 06kellyjac** [comment]
Thanks @jlesquembre would you mind if I add myself as a maintainer too?

**Comment by jlesquembre** [comment]
@06kellyjac go for it, better to have more maintainers :-)

**Comment by 06kellyjac** [comment]
Added :slightly_smiling_face:

File: pkgs/applications/networking/cluster/kube3d/default.nix [hunk 1]
```diff
@@ -2,16 +2,16 @@
 
 buildGoModule rec {
   pname = "kube3d";
-  version = "3.0.2";
-  k3sVersion = "1.18.6-k3s1";
+  version = "3.1.5";
+  k3sVersion = "1.18.9-k3s1";
```
Comment (jlesquembre) [inline pkgs/applications/networking/cluster/kube3d/default.nix:6]: Latest k3s version is 1.18.10, should we use that one?

**Review by jlesquembre — COMMENTED** [review]
I notice that k3d added completions for fish, can you add it too?
```
    for shell in bash zsh fish; do
```
Other than that, all looks good, tested it locally

**Comment by 06kellyjac** [comment]
Sure, I'll pop that in tomorrow morning

File: pkgs/applications/networking/cluster/kube3d/default.nix (hunk 1 above)
Comment (06kellyjac) [reply pkgs/applications/networking/cluster/kube3d/default.nix:6]: I've just been copying whatever value is in the released k3d binary, the image can be overriden with the `--image` flag when creating the cluster

**Comment by 06kellyjac** [comment]
image

I added it to the start of this PR as the feature was added in `3.0.2`

File: pkgs/applications/networking/cluster/kube3d/default.nix (hunk 1 above)
Comment (jlesquembre) [reply pkgs/applications/networking/cluster/kube3d/default.nix:6]: Fair enough, makes sense to align with the upstream version

File: pkgs/applications/networking/cluster/kube3d/default.nix [hunk 2]
```diff
@@ -22,11 +22,13 @@ buildGoModule rec {
   '';
 
   nativeBuildInputs = [ installShellFiles ];
+
+  # TODO: Move to enhanced installShellCompletion when in master: PR #83630
```
Comment (jlesquembre) [inline pkgs/applications/networking/cluster/kube3d/default.nix:26]: Isn't #83630 already merged?

Comment (06kellyjac) [reply pkgs/applications/networking/cluster/kube3d/default.nix:26]: It's not in master yet
image

Comment (jlesquembre) [reply pkgs/applications/networking/cluster/kube3d/default.nix:26]: right, my bad, it's only on staging

**Review by jlesquembre — APPROVED** [review]

**Comment by jlesquembre** [comment]
@06kellyjac I was trying to use marvin: https://github.com/timokau/marvin-mk2/blob/master/USAGE.md
but looks like the PR author needs to do it, can you try?
I tested it and looks good to me, but I don't have merge rights on nixpkgs.

**Comment by 06kellyjac** [comment]
Theres also a discourse thread for PRs that are reviewed but need a merger. I'd grab the link but I'm on mobile atm.

**Review by timokau — APPROVED** [review]
Thanks!

**Comment by timokau** [comment]
> Theres also a discourse thread for PRs that are reviewed but need a merger. I'd grab the link but I'm on mobile atm.

Marvin is intended to be a better replacement for that. It
- helps you find the initial review as well
- is higher signal, lower noise (this is mostly from a reviewer/"commiter" perspective since marvin can actually verify that there *was* a 3rd-party review)
- distributes the work better by assigning reviewers / "commiters".

Its still experimental though and will likely remain that way for quite a while.

**Comment by 06kellyjac** [comment]
Does marvin have a dashboard too or is it just good at only assigning tags if they're valid & replying well to incorrect commands?

**Comment by timokau** [comment]
I'm not sure what you mean. There is no dashboard. What information would you want displayed? If you want to find PRs to review, you can use this filter: https://github.com/NixOS/nixpkgs/labels/needs_reviewer

It usually does not reply to misformatted commands (it should eventually do that, but doesn't for now). The best source of "end user" documentation is probably [the bot's "profile"](https://github.com/apps/marvin-mk2) right now.

**Comment by 06kellyjac** [comment]
Yeah I was just wondering if it was adding good tags for filtering or if there was something else it was providing for nixpkgs pr observability that i didnt know about

I dont mean misformatted. I mean it told me when i couldn't change to awaiting merger and that it wanted the reviewer to post it. Which was helpful

**Comment by 06kellyjac** [comment]
If i put /m*rvin opt-in in the original PR description will it work or does it have to be a comment?

**Pull request merged**